### PR TITLE
Update BAP UI

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/13701

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [x] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Text and design for popup:

Screen Shot 2021-02-01 at 3 35 41 PM

**Note: due to technical limitations we were unable to render rounded borders and the bubble arrow:**

Screen Shot 2021-02-01 at 15 56 06

Text and design for alert:

Screen Shot 2021-02-01 at 3 34 57 PM

### Scenario 1: Non-JP Regions

- Open the browser in rewards staging environment.
- Open the rewards panel and claim the promotion.
- Wait for the promotion value to appear in the wallet balance on the panel.
- Close the rewards panel and open a new tab.
  - Verify that the BAP popup does not appear on the new tab page.
- Close the new tab page.
- Advance the system time to 2021-03-06.
- Open a new tab page.
  - Verify that the BAP alert does not appear on the new tab page.
- Close the new tab page.
- Advance the system time to 2021-03-13.
- Open a new tab page and open the rewards panel.
  - Verify that the wallet balance is not zero.

### Scenario 2: JP Regions

- Set the system region to JP.
- Open the browser in rewards staging environment.
- Open the new tab page.
  - Verify that the BAP popup does not appear (because the user's balance is zero).
- Close the new tab page.
- Open the rewards panel and claim the promotion.
- Wait for the promotion value to appear in the wallet balance on the panel.
- Close the rewards panel and open a new tab.
  - Verify that the BAP popup appears.
- Click the "Learn more" button on the popup.
  - Verify that the BAP alert appears in the new tab page.
- Close the BAP alert and the new tab page.
- Open a new tab.
  - Verify that the BAP popup does not appear.
- Close the new tab page.
- Advance the system time by 3 days.
- Open a new tab page.
  - Verify that the BAP popup is displayed.
- Close the new tab page.
- Advance the system time to 2021-03-06.
- Open the new tab page.
  - Verify that the BAP alert is displayed.
- Dismiss the BAP alert by clicking the "X" button and close the new tab page.
- Open a new tab.
  - Verify that the BAP alert is displayed.
- Click the "OK" button on the BAP alert.
- Close the tab.
- Open a tab.
  - Verify that the BAP alert is not displayed.
- Close the tab.
- Advance the system time by 3 days.
- Open a tab.
  - Verify that the BAP alert is displayed.
- Click the "OK" button on the BAP alert.
- Close the tab.
- Advance the system time to 2021-03-13.
- Open a new tab.
  - Verify that the BAP popup and the BAP alert are not displayed.
- Open the rewards panel.
  - Verify that zero is shown for the user's balance.

 ### Scenario 3: JP Regions (Promotions)

- Set the system region to JP.
- Advance the system time to 2021-03-13.
- Open the browser in rewards staging environment.
- Open the rewards panel.
- Verify that no grant promotions are displayed to the user.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -1489,6 +1489,22 @@ void RewardsServiceImpl::ClearState(const std::string& name) {
 bool RewardsServiceImpl::GetBooleanOption(const std::string& name) const {
   DCHECK(!name.empty());
 
+  if (name == ledger::option::kContributionsDisabledForBAPMigration) {
+    if (OnlyAnonWallet()) {
+      base::Time::Exploded cutoff_exploded;
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply components/brave_rewards/browser/rewards_service_impl.cc:1494]: Yep - it's not initialized properly. Thanks!

File: components/resources/brave_components_strings.grd [hunk 2]
```diff
@@ -955,6 +955,16 @@
...
       </message>
+      <!-- Rewards BAP Deprecation Notice -->
+      <message name="IDS_REWARDS_BAP_DEPRECATION_HEADER" desc="">
+        Brave Rewards in Japan is switching to a new system!
+      </message>
+      <message name="IDS_REWARDS_BAP_DEPRECATION_ALERT_TEXT" desc="">
+        The tipping feature will be unavailable from March 13th until April 13th. All BAPs will
+        be discontinued by April 13th. Please use your BAP by March 13th. Any BAP remaining in
+        your Brave browser after March 13th cannot be held or transferred by you or Brave, and
+        will be lost. A new and improved system awaits!
```
Comment (bsclifton) [inline components/resources/brave_components_strings.grd:966]: Matching text in comment: https://github.com/brave/internal/issues/747#issuecomment-764774254 should there be a new line between `will be lost.` and `A new and improved system awaits!`? If so, might need to be broken into two parts

Comment (zenparsing) [reply components/resources/brave_components_strings.grd:966]: The text at https://www.figma.com/file/9Pn7htrOeZsbOJ2NcxtxvR/bitFlyer-Promo?node-id=206%3A0 does not have a line break - @codybanxs can you confirm that the text layout in the design is what we want?

Comment (bsclifton) [reply components/resources/brave_components_strings.grd:966]: oh yeah - good call (with Figma link). I guess either works - maybe we can add test plan after which QA can look at, depending on which is the one they should compare against 🙂

**Comment by zenparsing** [comment]
Added two browsertests to cover code in `RewardsServiceImpl` that determines whether BAP is enabled - I modeled them on  some of the non-flaky rewards browsertests, so I'm hoping that they will be reliable.

File: components/resources/brave_components_strings.grd (hunk 2 above)
Comment (codybanxs) [reply components/resources/brave_components_strings.grd]: While the line break works okay on the mobile UI, it does not work well on the desktop oddly enough. Basically I did not put in a line break due to the length and flow of the copy across the board and to keep things visually consistent. If we do want to proceed with that I would need to tweak the container sizing abit which is not a problem.

**Review by emerick — APPROVED** [review]
LGTM

**Review by tmancey — APPROVED** [review]
LGTM++

**Comment by codybanxs** [comment]
looking in from a time expediency lens, this tooltip notice should suffice. One quick thought though @zenparsing is perhaps to use our existing blue tooltip for the phased welcome tour for new users but that might be out of reach for this. In any case this will work fine, great job!

**Comment by LaurenWags** [comment]
Testing of this on Nightly is blocked by https://github.com/brave/brave-browser/issues/14092. Discussed with @zenparsing so he is aware. Unable to get BAP so I can't check scenarios 2 and 3. cc @Miyayes @kjozwiak

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave	1.22.19 Chromium: 89.0.4389.40 (Official Build) nightly (x86_64)
Revision	2c2ed83cd507b23e4845edd09a7d1dfc727daf4b-refs/branch-heads/4389@{#602}
OS	macOS Version 10.15.7 (Build 19H512)
```

Verified test plan from https://github.com/brave/brave-core/pull/7698
Verified https://github.com/brave/brave-browser/issues/14092#issuecomment-777589672 also.

<details>
<summary>Scenario 1: Non-JP Regions</summary>

Checked with US/English and France/French

Confirmed able to claim promotion:

US/English | France/French
----- | -----
US1 | FR1
US2 | FR2


Confirmed BAP popup does not appear on NTP:

US/English | France/French
----- | -----
US3 | FR3


Confirmed BAP alert does not display on NTP when system date is set to 2021-03-06:

US/English | France/French
----- | -----
US4 | FR4


Confirmed wallet balance is not zero when system date is set to 2021-03-13:

US/English | France/French
----- | -----
US5 | FR5

</details>

**Comment by kjozwiak** [comment]
Moving this into `1.22.x` as the BAP work is being reverted via https://github.com/brave/brave-core/pull/8088. Once we have more information next week, we'll re-revert https://github.com/brave/brave-core/pull/8088 and move everything back into `1.21.x` and release `1.21.x - Release #2`.

**Pull request merged**